testDll/op2ext.test.cpp

```diff
@@ -0,0 +1,188 @@
+#include "op2ext.h"
+#include <gtest/gtest.h>
+#include <string_view>
+#include <fstream>
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+
+TEST(op2ext, GetGameDir_s) {
+	constexpr char Nonce = 255;
```

Comment:
I'm a bit confused by what is meant by Nonce

Reply:
It's a fixed known value so you can detect writes or non-writes to a field. In this case, it's being used to check that a null terminator is not written when there is no valid buffer to write to. It is also  used to detect when the null terminator is written, as without initialization to a known value, you can't be certain if a value was an uninitialized null, or an explicitly written null.